contact_manager.py

```diff
@@ -1,218 +1,281 @@
+        elif phone_no in self.provider.Idea:
+            self.provider.provider_name = "Idea"
+        elif phone_no in self.provider.Reliance:
+            self.provider.provider_name = "Reliance"
+        else:
+            self.provider.provider_name = "Others"
 
-    def modify_person(self, phone_no, fields):
+    def get_provider_name(self, phone_no):
+        self.set_provider_name(phone_no)
+        return self.provider.provider_name
+
+
+class Contacts:
+    def __init__(self, contact_list):
+        self.contact_list = contact_list
+
+    def add_contact(self, contact):
+        self.contact_list[contact.phone_no] = contact
+
+    def modify_contact(self, phone_no, fields):
         for key in fields.keys():
-            if key == 'name':
-                self.data[phone_no].name = fields[key]
-            elif key == 'phone_no':
-                self.data[phone_no].phone_no = fields[key]
-                self.data[fields[key]] = self.data.pop[phone_no]
-            elif key == 'email':
-                self.data[phone_no].email = fields[key]
-            elif key == 'street':
-                self.data[phone_no].addres.street = fields[key]
-            elif key == 'city':
-                self.data[phone_no].addres.city = fields[key]
-            elif key == 'state':
-                self.data[phone_no].addres.state = fields[key]
-            elif key == 'pin_code':
-                    self.data[phone_no].addres.pin_code = fields[key]
-
-    def delete_person(self, phone_no):
-        del self.data[phone_no]
-
-    def get_details(self, phone_no):
-        return self.data[phone_no]
+            if key == ContactDetails.NAME:
+                self.contact_list[phone_no].name = fields[key]
+            elif key == ContactDetails.PHONE_NO:
+                self.contact_list[phone_no].phone_no = fields[key]
+                self.contact_list[fields[key]] = self.contact_list.pop[phone_no]
+            elif key == ContactDetails.EMAIL:
+                self.contact_list[phone_no].email = fields[key]
+            elif key == ContactDetails.STREET:
+                self.contact_list[phone_no].addres.street = fields[key]
+            elif key == ContactDetails.CITY:
+                self.contact_list[phone_no].addres.city = fields[key]
+            elif key == ContactDetails.STATE:
+                self.contact_list[phone_no].addres.state = fields[key]
+            elif key == ContactDetails.PIN_CODE:
+                self.contact_list[phone_no].addres.pin_code = fields[key]
+
+    def delete_contact(self, phone_no):
+        del self.contact_list[phone_no]
+
+    def get_contact(self, phone_no):
+        return self.contact_list[phone_no]
 
     def get_provider(self, phone_no):
-        s = phone_no[0:4]
-        for key, values in Contacts.provider.items():
-            if s in values:
-                return key
-        return 'Others'
+        ProviderManager.get_provider_name(phone_no[0:4])
```

Comment:
read clearly difference between static and non static methods!! and their usage!!